### DISCUSSION TITLE
ts_rank weighting for sorting not working in Javascript with `textSearch`

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#14959] [Questions]
# Problem
I try to get my already weighted result column with `.textSearch` in javascript sorted by my weighted column. But my results are not the same as the ones from my SQL query. The comparison is below. It seems `.textSearch` is not considering my weighting.

# Question
How can I get the weighting to work with `.textSearch``

# What I've tried

## Create my weighted column
```sql
ALTER TABLE contacts 
ADD COLUMN
wtsv tsvector GENERATED ALWAYS AS   ( setweight ( to_tsvector ( 'simple' ,   coalesce ( first_name ,   '' )),   'A' )   ||   setweight ( to_tsvector ( 'simple' ,   coalesce ( last_name ,   '' )),   'B' )   ||   setweight ( to_tsvector ( 'simple' ,   coalesce ( notes ,   '' )),   'C' ))  STORED ;

CREATE  INDEX contacts_wtsv_idx ON  contacts USING  gin ( wtsv );   -- generate the index
```

## Get my weighted results SQL-Approach(which works)
```sql
select   *   from  contacts where  wtsv @@ to_tsquery ( REPLACE ( 'Bob' ,   ' ' ,   ':* & ' )   ||   ':*' )
order  by ts_rank ( wtsv ,   to_tsquery ( REPLACE ( 'Bob' ,   ' ' ,   ':* & ' )   ||   ':*' ))   desc ;
```

## Get my weighted results Javascript approach (The results are not the same as in SQL it seems like the weighting is not working)
```js
const searchText = "Bob" // Or "Bob Smith" or "Smith Box"
const searchQuery = searchText.trim().replace(" ", ":* & ") + ":*";

supabase.from("contacts").select("*", { count: "exact" }).textSearch("wtsv", `${searchQuery}`).order("wtsv", { ascending: false });
```

**Comment by mansueli** [comment]
The main difference between both calls is that you are not using `order by` in the supabase/postgrest call. So that will yield to different results. 

You can always encapsulate this call in an RPC call (postgres Function) to use the same query as in SQL.

**Reply by bennik88** [reply]
@mansueli Thank you for taking a look at my question. I'm sorry but I forgot to include my `order("wtsv", { ascending: false })` when writing the question. In my test, I've included it for sure. That's why I think the order by Javascript version is not working correctly. And it's giving different results with the `order` method call. I know I can use RPC but the point of using supabase is to also make use of the included API instead of writing everything yourself. As therefore i can use the `{ count: "exact" }` functionality and the `.range(startRange, endRange);`.

So why is the `textSearch` not working with weighted columns?

**Reply by bennik88** [reply]
Also if I'm not creating a RPC function because therefore I would face a lot of new problems. One would be getting the count back without two reads of the whole table. Second I would have to hardcode all return columns into the function which is not good in a prototyping stage when the schema is changing a lot. It wouldn't be very performant.

**Comment by mansueli** [comment]
That's still a different ordering way:

`.order("wtsv", { ascending: false });` is not the same as `order  by ts_rank ( wtsv ,   (...))   desc ;`

You can use dynamic functions if you need to specify different columns. Order by ts_rank is not supported by postgREST. 
You would need to add that into a view, then query the view with this already ranked to get the same results.
